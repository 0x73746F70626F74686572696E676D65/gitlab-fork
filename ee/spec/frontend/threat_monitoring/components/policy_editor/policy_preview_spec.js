import { GlTabs } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import PolicyPreviewHuman from 'ee/threat_monitoring/components/policy_editor/policy_preview_human.vue';
import PolicyPreview from 'ee/threat_monitoring/components/policy_editor/policy_preview.vue';

describe('PolicyPreview component', () => {
  let wrapper;

  const findTabs = () => wrapper.findComponent(GlTabs);
  const findPolicyPreviewHuman = () => wrapper.findComponent(PolicyPreviewHuman);

  const factory = ({ propsData } = {}) => {
    wrapper = shallowMount(PolicyPreview, {
      propsData: {
        ...propsData,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('with policy description', () => {
    const policyDescription = '<strong>bar</strong><br><div>test</div><script></script>';

    beforeEach(() => {
      factory({
        propsData: {
          policyYaml: 'foo',
          policyDescription,
        },
      });
    });

    it('renders policy preview tabs', () => {
      expect(findTabs().element).toMatchSnapshot();
    });

    it('renders the policy preview human', () => {
      expect(findPolicyPreviewHuman().props('policyDescription')).toBe(policyDescription);
    });

    it('renders the first tab', () => {
      expect(findTabs().attributes().value).toEqual('0');
    });

    describe('initial tab', () => {
      it('selects initial tab', () => {
        factory({
          propsData: {
            policyYaml: 'foo',
            policyDescription: 'bar',
            initialTab: 1,
          },
        });
        expect(findTabs().attributes().value).toEqual('1');
      });
    });
  });

  describe('without policy description', () => {
    beforeEach(() => {
      factory({
        propsData: {
          policyYaml: 'foo',
        },
      });
    });
  });
});
