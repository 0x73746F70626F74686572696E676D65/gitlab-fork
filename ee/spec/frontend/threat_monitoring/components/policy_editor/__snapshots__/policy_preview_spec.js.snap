// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyPreview component with policy description renders policy preview tabs 1`] = `
<gl-tabs-stub
  contentclass="gl-pt-0"
  queryparamname="tab"
  theme="indigo"
  value="0"
>
  <gl-tab-stub
    title="Rule"
    titlelinkclass=""
  >
    <policy-preview-human-stub
      class="gl-border-t-none! gl-rounded-top-left-none gl-rounded-top-right-none"
      policydescription="<strong>bar</strong><br><div>test</div><script></script>"
    />
  </gl-tab-stub>
   
  <gl-tab-stub
    title=".yaml"
    titlelinkclass=""
  >
    <pre
      class="gl-bg-white gl-rounded-top-left-none gl-rounded-top-right-none gl-border-t-none"
    >
      foo
    
    </pre>
  </gl-tab-stub>
</gl-tabs-stub>
`;
